### PR TITLE
move zbxapi module to a new os_zabbix role

### PR DESCRIPTION
**Pull request by detiber**
- cleans up repo root a bit

### CODE REVIEW

**Comment by detiber** [comment]
@kwoodson @twiest I found the top level library folder a bit distracting, this just moves the zbxapi.py module contained there to a new role.

File: playbooks/adhoc/noc/get_zabbix_problems.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@
 - name: 'Get current hosts who have triggers that are alerting by trigger description'
   hosts: localhost
   gather_facts: no
+  roles:
+    - os_zabbix
   tasks:
```
Comment (twiest) [inline playbooks/adhoc/noc/get_zabbix_problems.yml]: If you're going to rely on the role to pull in the module, then you should change "tasks" to "post_tasks" to make it clear that it'll run after roles (more clarity than anything IMO).

**Comment by twiest** [comment]
@detiber did you test that this is still working? I didn't know you could pull in ansible modules globally by simply including a role. I thought only the role that includes the module could access the module (like some sort of namespacing).

If this works, then it's a very cool way to pull in custom modules.

**Comment by detiber** [comment]
@twiest I tested it as best as I could (since I can't access the zabbix host), but the module gave the same error as in the previous location.  Also, I initially used this approach with the openshift_facts module before adding a tasks/main.yml file.

Once the role has been included any modules (as well as other resources available in the role) are available to later roles and tasks.

**Comment by detiber** [comment]
@twiest changed tasks to pre_tasks

Also, further clarification on when modules are available, I know they are available to later roles and tasks within a play, I am not sure that they are available to later plays in the same playbook.

**Comment by twiest** [comment]
@detiber This is great stuff, thanks! This is yet another pattern that we will likely adopt on the ops team. 

:+1:

**Pull request merged**